examples/imagenet/train_resnet_imagenet1k.py

```diff
@@ -0,0 +1,297 @@
+                    default='1dur')
+
+# Save checkpoint arguments
+parser.add_argument('--save_checkpoint_dir',
+                    help='Directory in which to save model checkpoints',
+                    type=str,
+                    default='checkpoints/{run_name}')
+parser.add_argument('--checkpoint_interval', help='Frequency to save checkpoints', type=str, default='1ep')
+
+# Load checkpoint arguments, assumes resuming the previous training run instead of fine-tuning
+parser.add_argument('--load_checkpoint_path', help='Path to the checkpoint to load', type=str)
+
+# Recipes
+parser.add_argument('--recipe_name',
+                    help='Either "mild", "medium" or "spicy" in order of increasing training time and accuracy',
+                    type=str,
+                    choices=['mild', 'medium', 'spicy'])
+
+# Logger parameters: progress bar logging is used by default
+# Only has Weights and Biases option to reduce the number of arguments. Other loggers can be substituted in the script
+parser.add_argument('--wandb_logger', help='Whether or not to log results to Weights and Biases', action='store_true')
+parser.add_argument('--wandb_entity', help='WandB entity name', type=str)
+parser.add_argument('--wandb_project', help='WandB project name', type=str)
+parser.add_argument('--wandb_run_name', help='WandB run name', type=str)
+
+# Trainer arguments
+parser.add_argument('--run_name', help='Name of the training run used for checkpointing and other logging', type=str)
+parser.add_argument('--seed', help='Random seed', type=int, default=17)
+parser.add_argument('--max_duration',
+                    help='Duration to train specified as a Time string',
+                    type=Time.from_timestring,
+                    default='90ep')
+parser.add_argument('--eval_interval',
+                    help='How frequently to run evaluation on the validation set specified as a Time string',
+                    type=Time.from_timestring,
+                    default='1ep')
+
+# Local storage checkpointing
+args = parser.parse_args()
+
+
+def _main():
+
+    # Divide batch sizes by number of devices if running multi-gpu training
+    if dist.get_world_size():
+        args.train_batch_size = args.train_batch_size // dist.get_world_size()
+        args.eval_batch_size = args.eval_batch_size // dist.get_world_size()
+
+    IMAGENET_CHANNEL_MEAN = (0.485, 0.456, 0.406)
+    IMAGENET_CHANNEL_STD = (0.229, 0.224, 0.225)
+
+    # Train dataset
+    logging.info('Building train dataloader')
+    train_transforms = transforms.Compose([
+        transforms.RandomResizedCrop(args.train_crop_size, scale=(0.08, 1.0), ratio=(0.75, 4.0 / 3.0)),
+        transforms.RandomHorizontalFlip(),
+    ])
+    train_dataset = ImageFolder(os.path.join(args.data_dir, 'train'), train_transforms)
+    # Nifty function to instantiate a PyTorch DistributedSampler based on your hardware setup
+    train_sampler = dist.get_sampler(train_dataset, drop_last=True, shuffle=True)
```

Comment:
Do we need this? I don't have this in the science repo but also wouldn't be _that_ surprised if I've been sampling wrong this whole time.